plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/FacetInstallDelegate.java

```diff
@@ -89,15 +83,30 @@ public static void installAppEngineRuntime(IFacetedProject project, IProgressMon
     }
   }
 
+  /**
+   * Checks to see if <code>facetedProject</code> has the App Engine facet installed. If not, it installs
+   * the App Engine facet.
+   *
+   * @param facetedProject the workspace faceted project????????/
+   * @param monitor the progress monitor
+   * @throws CoreException
+   */
   public static void installAppEngineFacet(IFacetedProject facetedProject, IProgressMonitor monitor)
       throws CoreException {
-    IFacetedProjectWorkingCopy workingCopy = facetedProject.createWorkingCopy();
     IProjectFacet appEngineFacet = ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID);
     IProjectFacetVersion appEngineFacetVersion = appEngineFacet.getVersion(AppEngineStandardFacet.VERSION);
-    workingCopy.addProjectFacet(appEngineFacetVersion);
-    workingCopy.commitChanges(monitor);
+
+    // TODO: fix - this something cause the following error-------------------------------
+    if (!facetedProject.hasProjectFacet(appEngineFacet)) {
```

Comment:
If they have this facet, it's a different version and should be handled by a VERSION_CHANGE?